File: p004/p004.py
def isPalindrome(s):
    palindrome = True
    for i in range(len(s) // 2):
        if s[i] != s[len(s) - i -1]:
            palindrome = False
            break
    return palindrome

def largestPalindromeProduct(n):
    maxN = int("9" * n)
    minN = int("1" + "0" * (n - 1))
    s = 0
    for i in range(maxN, minN, -1):
        for j in range(i, minN, -1):
            product = i * j
            if isPalindrome(str(product)):
                if product > s:
                    s = product
                break
    return s
if __name__ == "__main__":
    print(largestPalindromeProduct(3))

File: p004/p004.js
const isPalindrome = s => {
  let palindrome = true,
    lengthS = s.length;
  for (let i = 0; i <= lengthS / 2; i++) {
    if (s[i] !== s[lengthS - i - 1]) {
      palindrome = false;
      break;
    }
  }
  return palindrome;
};
function largestPalindromeProduct(n) {
  // Good luck!
  let maxN = parseInt("9".repeat(n));
  let minN = parseInt("1".concat("0".repeat(n - 1)));
  let s = 0;
  for (let i = maxN; i > minN; i--) {
    for (let j = i; j > minN; j--) {
      let product = i * j;
      if (isPalindrome(product.toString())) {
        if (product > s) s = product;
        break;
      }
    }
  }
  return s;
}

largestPalindromeProduct(3);
